toolkit.lisp

```diff
@@ -65,18 +65,31 @@
                (warn "No documentation for ~(~a~) ~a." type symb)))))
 
 (defmacro define-docs (&body expressions)
-  `(progn
-     ,@(loop for expr in expressions
-             for length = (length expr)
-             for type = (if (< 2 length) (first expr) 'function)
-             for var = (if (< 2 length) (rest (butlast expr)) (butlast expr))
-             for doc = (car (last expr))
-             collect `(setf ,(funcall (documentation-translator type) var) ,doc))))
+  (multiple-value-bind (formatter expressions)
+      (if (eq :formatter (first expressions))
+          (values (second expressions) (cddr expressions))
+          (values 'basic-docstring-formatter expressions))
```

Comment:
This whole thing seems subpar to me. I would rather approach it in such a way that you can use an arbitrary amount of keyword arguments at the beginning of a docs definition, which are then passed as initargs to the formatter.

Reply:
This can be done.